test/commandLine/commandLineTest.js

```diff
@@ -0,0 +1,609 @@
+      done()
+    })
+  })
+
+  it('should change the app to put it into dev mode, disable validator, and detach validator ("-brd")', function (done) {
+    const testApp = fork(path.join(appDir, 'app.js'), ['-brd'], {'stdio': ['pipe', 'pipe', 'pipe', 'ipc']})
+
+    testApp.on('message', params => {
+      assert.equal(params.nodeEnv, 'development')
+      assert.equal(params.htmlValidator.enable, false)
+      assert.equal(params.htmlValidator.separateProcess, true)
+      done()
+    })
+  })
+
+  it('should change the app to put it into production mode, disable validator, and detach validator ("-prb")', function (done) {
+    const testApp = fork(path.join(appDir, 'app.js'), ['-prb'], {'stdio': ['pipe', 'pipe', 'pipe', 'ipc']})
+
+    testApp.on('message', params => {
+      assert.equal(params.nodeEnv, 'production')
+      assert.equal(params.htmlValidator.enable, false)
+      assert.equal(params.htmlValidator.separateProcess, true)
+      done()
+    })
+  })
+
+  it('should change the app to put it into production mode, disable validator, and detach validator ("-rbp")', function (done) {
+    const testApp = fork(path.join(appDir, 'app.js'), ['-prb'], {'stdio': ['pipe', 'pipe', 'pipe', 'ipc']})
+
+    testApp.on('message', params => {
+      assert.equal(params.nodeEnv, 'production')
+      assert.equal(params.htmlValidator.enable, false)
+      assert.equal(params.htmlValidator.separateProcess, true)
+      done()
+    })
+  })
+
+  it('should change the app to put it into production mode, should not enable validator, and detach validator ("-bhp")', function (done) {
+    const testApp = fork(path.join(appDir, 'app.js'), ['-bhp'], {'stdio': ['pipe', 'pipe', 'pipe', 'ipc']})
+
+    testApp.on('message', params => {
+      assert.equal(params.nodeEnv, 'production')
+      assert.equal(params.htmlValidator.enable, false)
+      assert.equal(params.htmlValidator.separateProcess, true)
+      done()
+    })
+  })
+
+  it('should change the app to put it into production mode, should not enable validator, and attach validator ("-hpa")', function (done) {
+    const testApp = fork(path.join(appDir, 'app.js'), ['-hpa'], {'stdio': ['pipe', 'pipe', 'pipe', 'ipc']})
+
+    testApp.on('message', params => {
+      assert.equal(params.nodeEnv, 'production')
+      assert.equal(params.htmlValidator.enable, false)
+      assert.equal(params.htmlValidator.separateProcess, false)
+      done()
+    })
+  })
+
+  it('should change the app to put it into production mode, disable validator, and attach validator ("-par")', function (done) {
+    const testApp = fork(path.join(appDir, 'app.js'), ['-par'], {'stdio': ['pipe', 'pipe', 'pipe', 'ipc']})
+
+    testApp.on('message', params => {
+      assert.equal(params.nodeEnv, 'production')
+      assert.equal(params.htmlValidator.enable, false)
+      assert.equal(params.htmlValidator.separateProcess, false)
+      done()
+    })
+  })
+
+  /* check mutually exclusive flags  */
```

Comment:
Similar to the combinations tests we don't need to check `--prod --dev` if we already checked `--production-mode --development-mode`

Condense these mutually exclusive tests down to unique combinations of flags.